guard/src/rules/eval_tests.rs

```diff
@@ -1743,6 +1743,108 @@ Resources:
     Ok(())
 }
 
+#[test]
+fn is_bool() -> Result<()> {
+    let rule_str = r###"
+    rule check_is_bool{
+        foo is_bool
+    }
+    "###;
+
+    let resources_str = r###"
+    {
+        foo: false
+    }
+    "###;
+
+    let value = PathAwareValue::try_from(resources_str)?;
+    let rules_file = RulesFile::try_from(rule_str)?;
+    println!("{:?}", rules_file);
+    let mut eval = root_scope(&rules_file, &value)?;
+    let status = eval_rules_file(&rules_file, &mut eval)?;
+    assert_eq!(status, Status::PASS);
+
+    let resources_str = r###"
+    {
+        foo: "false"
```

Comment:
When `foo` is empty in the template, the underlying code panics with a non-zero error code, we might have to handle that scenario to fail the rule evaluation itself.

`bool-template.yml `

```
bar: test
foo:
```

Similarly, when we pass a string value in a YAML formatted template, we get a panic instead of a failure in eval.
`bool-template.yml `
```
bar: test
foo: test
```

I checked with similar rule for is_list and we get only an eval failure without panic. So we might need to resolve the underlying error.

```
bool-template.yml Status = FAIL
FAILED rules
bool-rule.guard/check_is_bool    FAIL
---
Evaluation of rules bool-rule.guard against data bool-template.yml
--
thread 'main' panicked at 'internal error: entered unreachable code', guard/src/commands/validate/common.rs:537:43
```